### PR TITLE
feat: progress flag for CloudVolume, Storage

### LINKED ISSUE #20: Replace queue.join()
ThreadedQueue uses queue.join() which blocks until the entire queue is done. This prevents on demand error checking and status updates _a la_ tqdm. This can probably be replaced with a while loop and sleep that will process the queue in smaller batches before checking up on things.

### PR DESCRIPTION
**Pull request by william-silversmith**
```
CloudVolume(..., progress=True) # Shows progress bars
Storage(..., progress=True) # shows progress bars
```

ThreadedQueue now checks every 0.1 seconds for exceptions and updates
a progress indicator. Storage now takes advantage of this for
multithreaded uploads and downloads. CloudVolume now supports a
vol.progress flag that switches tqdm output on and off, default False.

Updated some comments for Storage as well.

Resolves #20

### CODE REVIEW

**Comment by william-silversmith** [comment]
Something curious about this. The main thread sleepy spin lock I implemented seems to cause an increase of about 1-1.5 sec over a simple `self._queue.join()` when downloading test_v0. 

That is about 16-17s for old implementation vs 17.3-18.2 for this version. Increasing the sleep time doesn't seem to make much of a difference.

**Comment by william-silversmith** [comment]
Some further testing showed that this effect is within the margin of variation. A more controlled experiment is needed to determine if it's real.

**Comment by william-silversmith** [comment]
@davidbuniat ran into an issue with CloudVolume where he was running CloudVolume inside of a multithreaded environment. He ran into a multithreading issues with `tqdm`. I wonder if this would fix that with `progress=False`

**Comment by davidbuniat** [comment]
my issue connected with tqdm on this branch seems resolved

File: cloudvolume/cloudvolume.py [hunk 1]
```diff
@@ -640,8 +652,10 @@ def upload_image(self, img, offset):
     if str(self.dtype) != str(img.dtype):
       raise ValueError('The uploaded image data type must match the volume data type. volume: {}, image: {}'.format(self.dtype, img.dtype))
 
+    iterator = tqdm(self._generate_chunks(img, offset), disable=(not self.progress), desc='rechunking image')
```
Comment (wongwill86) [inline cloudvolume/cloudvolume.py]: For consistence, you should match the order of the args (in `filteiter` you did `desc` then `disable`).
also capital R in `rechunking image`

File: cloudvolume/threaded_queue.py [hunk 2]
```diff
@@ -175,30 +179,62 @@ def _consume_queue_execution(self, fn):
...
+      desc = progress
+
+    last = self._inserted
+    with tqdm(total=self._inserted, disable=(not progress), desc=desc) as pbar:
+      # Allow queue to consume, but check up on
+      # progress and errors every tenth of a second
+      while not self._queue.empty():
+        size = self._queue.qsize()
+        delta = last - size
+        if delta != 0: # We should crash on negative numbers
```
Comment (wongwill86) [inline cloudvolume/threaded_queue.py]: how is it possible to get negative numbers?

Comment (william-silversmith) [reply cloudvolume/threaded_queue.py]: Programming errors when you are changing the code.

Comment (william-silversmith) [reply cloudvolume/threaded_queue.py]: Also if someone interferes with self._inserted

**Pull request merged**